includes/api/class-give-api.php

```diff
@@ -1849,40 +1850,51 @@ public function process_api_key( $args ) {
 	/**
 	 * Generate new API keys for a user
 	 *
+	 * @param int     $user_id    User ID the key is being generated for.
+	 * @param boolean $regenerate Regenerate the key for the user.
+	 *
 	 * @access public
 	 * @since  1.1
 	 *
-	 * @param int     $user_id    User ID the key is being generated for
-	 * @param boolean $regenerate Regenerate the key for the user
-	 *
 	 * @return boolean True if (re)generated succesfully, false otherwise.
 	 */
 	public function generate_api_key( $user_id = 0, $regenerate = false ) {
 
+		// Bail out, if user doesn't exists.
 		if ( empty( $user_id ) ) {
 			return false;
 		}
 
 		$user = get_userdata( $user_id );
 
+		// Bail Out, if user object doesn't exists.
 		if ( ! $user ) {
 			return false;
 		}
 
+		// Flush Rewrite Rules for Give API to work for the first time.
+		flush_rewrite_rules();
```

Comment:
@mehul0810 @DevinWalker use `flush_rewrite_rules` here is not good. Can we use this on plugin activate and deactivation?
For ref: https://codex.wordpress.org/Function_Reference/flush_rewrite_rules